#187300153 Admin Account Disable and Activate

musabehonore opened the pull request:
# What does this PR do?
This PR implements the functionality for Admin to disable and activate any user account on the app.

# Description of Task to be completed?
An admin will be able to disable and activate any user's account and send them an email explaining the reasons for the suspension.
# How should this be manually tested?
### For testing only
npm install
npm run test
### For running the app
npm run install
npm run demigrate
npm run migrate
npm run seed
npm run dev

# Any background context you want to provide?
No background feature added.

# What are the relevant pivotal tracker/Trello stories?
-Story ID: #187300153

fab-ryan commented on src/controllers/loginController.ts:
```diff
@@ -17,16 +17,20 @@ const loginController = async (req: Request, res: Response) => {
       );
     }
+
+    if (user.status !== 'Active') {
```
please use boolean not a string

fab-ryan commented on src/controllers/userStatusController.ts:
```diff
@@ -0,0 +1,94 @@
+) => {
+  try {
+    const { id } = req.params;
+    const { action, reasons } = req.body;
+    const user = await Database.User.findOne({ where: { id } });
+    if (!user) {
+      return sendResponse<null>(res, 404, null, 'User not found');
+    }
+    if (user.status === statuses[action]) {
```
you don't need actions can be vice versa true or false

fab-ryan commented on src/database/migrations/20240411085039-add_auth_columns_to_users.js:
```diff
@@ -16,11 +16,17 @@ module.exports = {
       defaultValue: false,
     });
+    await queryInterface.addColumn('users', 'status', {
+      type: Sequelize.STRING,
+      allowNull: false,
+      defaultValue: 'Active',
```
please if you have new task that require to add new column please create it's own migration

fab-ryan reviewed (changes requested):
work on those commets

fab-ryan commented on src/routers/userRoute.ts:
```diff
@@ -45,5 +46,6 @@ router.patch(
 );
 
 router.get('/users/verify-email/:token', verifyEmail);
+router.post('/users/:id/accountStatus', isAuthenticated, isAdmin, userStatus);
```
we have naming conversation of api like if verifyEmail should be verify-email or account-status

fab-ryan reviewed (changes requested):
change that endpoint naming

musabehonore committed:
feat(suspend account): disable and activate an account

suspend an account only if admin
use nodemailer to send the emails explaining the reasons for the act
users can no longer login on the app when suspended
and vice versa when activating the account again
document the endpoint
test the endpoints

[#187300153]
